### PR TITLE
Theme-ability progress

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: -we want storybook to be theme-able / customisable-

## What I did
I changed more components to use emotion for styling, I also modified some components to prepare for customisation/theming ability.

It also includes a new mobile layout.

So I migrated everything over to use emotion.. surprise. It's basically because of this:
https://github.com/paypal/glamorous/issues/419

Also I was running into a strange snapshot mismatch issue. Hopefully emotion doesn't have this issue. 🤞 

## How to test
Run the examples, see chromatic diffs

### CODE REVIEW

File: lib/ui/src/modules/ui/components/stories_panel/index.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
-import Header from '../header';
+import glamorous from 'glamorous';
+
+import { Header } from '@storybook/components';
 import Stories from './stories_tree';
 import TextFilter from './text_filter';
 
-const scrollStyle = {
-  height: 'calc(100vh - 105px)',
-  marginTop: 10,
```
Comment (Hypnosphi) [inline lib/ui/src/modules/ui/components/stories_panel/index.js]: Looks like you've lost this margin

Comment (ndelangen) [reply lib/ui/src/modules/ui/components/stories_panel/index.js]: No just moved it to an other wrapper, or did you find something mis-aligned?

**Comment by ndelangen** [comment]
@Hypnosphi I'm kind of at a loss why the snapshots do not match up.. Any suggestions?

**Comment by ndelangen** [comment]
The test coverage increase is fake news btw, I found out the /dist/ files were included, and I removed them from coverage.

File: lib/ui/src/modules/ui/components/stories_panel/index.js (hunk 1 above)
Comment (ndelangen) [reply lib/ui/src/modules/ui/components/stories_panel/index.js]: Fixed

**Comment by tmeasday** [comment]
Pity there isn't a good way to look at all the changed stories at once @ndelangen 🤔 

Code-wise, looks great. Did the font color change slightly?

**Review by tmeasday — APPROVED** [review]

**Comment by ndelangen** [comment]
The TeamCity official example is a pretty good indicator.

Indeed the main colour changed a bit, I'll work on a follow-up PR that consolidates the UI, and implements theming in more places.

File: addons/knobs/src/react/WrapStory.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
+/* eslint no-underscore-dangle: 0 */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { polyfill } from 'react-lifecycles-compat';
```
Comment (shilman) [inline addons/knobs/src/react/WrapStory.js]: What does this have to do with theming? Can we move to another PR?

File: addons/viewport/src/manager/components/Panel.js [hunk 3]
```diff
@@ -170,11 +170,6 @@ export class Panel extends Component {
       this.iframe.style.height = viewport.styles.width;
       this.iframe.style.width = viewport.styles.height;
     }
-
-    // Always make parent's height equals iframe
-    if (this.iframe.parentElement) {
```
Comment (shilman) [inline addons/viewport/src/manager/components/Panel.js]: Why delete this???

File: lib/components/README.md [hunk 4]
```diff
@@ -11,4 +11,4 @@
 
 Storybook Components is a React UI components collection used by the UI of Storybook and Addons.
 
-All components use [`glamorous`](https://glamorous.rocks) for styling.
+All components use [`glamorous`](https://styled.rocks) for styling.
```
Comment (shilman) [inline lib/components/README.md]: https://emotion.sh/

**Review by shilman — CHANGES_REQUESTED** [review]
Looks great! Minor comments inline 👍

File: addons/viewport/src/manager/components/Panel.js (hunk 3 above)
Comment (ndelangen) [reply addons/viewport/src/manager/components/Panel.js]: Because I changed the way the overflow works for the preview

File: lib/components/src/layout/desktop.js [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+    saveSizes(sizes);
+    return sizes;
+  } catch (e) {
+    saveSizes(sizes);
+    return sizes;
+  }
+};
+
+class Layout extends Component {
+  constructor(props) {
```
Comment (Keraito) [inline lib/components/src/layout/desktop.js]: A decent amount of other components in this PR have their constructor and method bindings replaced by class properties and fields, while a lot have not. Would it be wise to choose 1 and stick with it throughout the project?

File: examples/html-kitchen-sink/package.json [hunk 6]
```diff
@@ -1,17 +1,17 @@
 {
   "name": "html-kitchen-sink",
   "version": "4.0.0-alpha.7",
+  "private": true,
   "description": "",
+  "keywords": [],
+  "license": "ISC",
```
Comment (Keraito) [inline examples/html-kitchen-sink/package.json]: Is there a reason this is the only subproject with a `ISC` license? From what I could find, [there doesn't seem to be a difference](https://gist.github.com/indexzero/10602128), but just curious why one folder is differently licensed.

**Review by Keraito — COMMENTED** [review]
Refactor wise looks great as far as I could tell! Just some small remarks or questions.

**Review by Hypnosphi — APPROVED** [review]

File: lib/components/src/layout/desktop.js (hunk 5 above)
Comment (ndelangen) [reply lib/components/src/layout/desktop.js]: I tend to write components using modules and module scope function and less with component instance function. That's just my coding style. (composition over inheritance)

Though I do believe that in a perfect world a codebase would be perfectly uniform, in reality code is messy for lots of reasons. I'd rather have people contribute and do a bit of gradual cleanup afterwards.

I personally think that code uniformity is highly overrated. It's very easy to be opinionated about and loose valuable time controlling petty details.

So to answer your question:
Yes it would be wise, but it's not something I'm willing to enforce, because there is a trade-off with getting new developers on board whose opinion and expertise is going to be different. I just apply the boy-scout rules: leave the campsite better then it was.

File: addons/a11y/src/components/Report/Elements.js [hunk 7]
```diff
@@ -1,31 +1,31 @@
...
-  },
-  target: {
-    borderBottom: '1px solid rgb(130, 130, 130)',
-    width: '100%',
-    display: 'inline-block',
-    paddingBottom: '4px',
-    marginBottom: '4px',
-  },
-};
+const Item = styled('li')({
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply addons/a11y/src/components/Report/Elements.js]: Because the `li` actually has user-agent styling applied to it, the `ol` doesn't as much.

File: examples/html-kitchen-sink/package.json (hunk 6 above)
Comment (ndelangen) [reply examples/html-kitchen-sink/package.json]: No there's no reason, the only reason this is in the PR, is because I auto-sort `package.json` files.

Please feel free to change it to MIT.
